router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -1,403 +1,355 @@
-func (b *MarketoBulkUploader) GetUploadStats(UploadStatsInput common.GetUploadStatsInput) common.GetUploadStatsResponse {
-	var jobsURL string
+func (b *MarketoBulkUploader) Poll(pollInput common.AsyncPoll) common.PollStatusResponse {
+	importId := pollInput.ImportId
 
-	if UploadStatsInput.FailedJobURLs != "" {
-		jobsURL = UploadStatsInput.FailedJobURLs
-	} else {
-		jobsURL = UploadStatsInput.WarningJobURLs
-	}
+	marketoResponse, apiError := b.apiService.PollImportStatus(importId)
 
-	parameters := UploadStatsInput.Parameters
-	importId := gjson.GetBytes(parameters, "importId").String()
-	csvHeaders := gjson.GetBytes(parameters, "metadata.csvHeader").String()
-	payload := b.generateFailedPayload(UploadStatsInput.ImportingList, importId, b.destName, csvHeaders)
-	if payload == nil {
-		return common.GetUploadStatsResponse{
-			StatusCode: 500,
-		}
-	}
+	b.logger.Debugf("[Async Destination Manager] Marketo Poll Response: %v", marketoResponse)
 
-	failedBodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(b.transformUrl+jobsURL, payload, b.timeout)
-	if statusCode != 200 {
-		return common.GetUploadStatsResponse{
-			StatusCode: statusCode,
+	if apiError != nil {
+		if apiError.Category == "RefreshToken" {
+			return common.PollStatusResponse{StatusCode: 500, Error: apiError.Message, Complete: false}
 		}
-	}
-	var failedJobsResponse common.GetUploadStatsResponse
-	err := json.Unmarshal(failedBodyBytes, &failedJobsResponse)
-	if err != nil {
-		b.logger.Errorf("Error in Unmarshalling Failed Jobs Response: %v", err)
-		return common.GetUploadStatsResponse{
-			StatusCode: 500,
+		switch apiError.StatusCode {
+		case 500:
+			return common.PollStatusResponse{StatusCode: int(apiError.StatusCode), Error: apiError.Message, Complete: false}
+		case 400:
+			return common.PollStatusResponse{StatusCode: int(apiError.StatusCode), Error: apiError.Message, Complete: true, HasFailed: true}
+		case 429:
+			return common.PollStatusResponse{StatusCode: int(apiError.StatusCode), Error: apiError.Message, Complete: false}
 		}
+
 	}
 
-	if failedJobsResponse.Error != "" {
-		b.logger.Errorw("[Batch Router] Failed to fetch status for",
-			lf.DestinationType, "MARKETO_BULK_UPLOAD",
-			"body", string(failedBodyBytes),
-			lf.Error, failedJobsResponse.Error,
-		)
-		return common.GetUploadStatsResponse{
+	// Check if the response is empty
+	if len(marketoResponse.Result) == 0 {
+		return common.PollStatusResponse{
 			StatusCode: 500,
+			Complete:   false,
+			HasFailed:  true,
+			Error:      "No result found in the API response",
 		}
 	}
 
-	failedReasons := failedJobsResponse.Metadata.FailedReasons
-	warningReasons := failedJobsResponse.Metadata.WarningReasons
+	batchStatus := marketoResponse.Result[0].Status
+	hasFailed := marketoResponse.Result[0].NumOfRowsFailed > 0
+	hasWarning := marketoResponse.Result[0].NumOfRowsWithWarning > 0
+
+	coreURl := fmt.Sprintf("https://%s.mktorest.com/bulk/v1/leads/batch/%s", b.destinationConfig.MunchkinId, importId)
```

Comment:
We should probably list all the urls in one place.. probably types.go and re use things from over there.

Reply:
Right now we have only core url which is majorly used for other urls as well with string formatting, and core url itself is generated using destination config parameters will refactor this url generation in a better way using utils for now lets keep this as a p1 backlog